Cdk pipeline modern api typescript (#295)
aws-samples/aws-cdk-intro-workshop: workshop/content/20-typescript/70-advanced-topics/200-pipelines/5000-test-actions.md

        // PIPELINE CODE HERE...

        const deploy = new WorkshopPipelineStage(this, 'Deploy');
        const deployStage = pipeline.addStage(deploy);

        deployStage.addPost(
            new CodeBuildStep('TestViewerEndpoint', {
                projectName: 'TestViewerEndpoint',
                envFromCfnOutputs: {

>                     ENDPOINT_URL: //TBD

Reviewer: ?

                },
                commands: [
                    'curl -Ssf $ENDPOINT_URL'
                ]
            }),

            new CodeBuildStep('TestAPIGatewayEndpoint', {
                projectName: 'TestAPIGatewayEndpoint',
                envFromCfnOutputs: {

>                     ENDPOINT_URL: //TBD

Reviewer: ?

Author: It comes later in the steps. Look at the diff from before :)

                },
                commands: [
                    'curl -Ssf $ENDPOINT_URL',

>                     'curl -Ssf $ENDPOINT_URL',
>                     'curl -Ssf $ENDPOINT_URL'

Reviewer: ```suggestion
                    'curl -Ssf $ENDPOINT_URL/hello',
                    'curl -Ssf $ENDPOINT_URL/test'
```

                ]
            })
        )
    }
}
{{</highlight>}}

> We add deployment post step via `deployStage.addPost(...)` from CDK Pipelines. We add two actions to our deployment stage that test our TableViewer endpoint and our APIGateway endpoint respectively.

Reviewer: ```suggestion
We add post-deployment steps via `deployStage.addPost(...)` from CDK Pipelines. We add two actions to our deployment stage: to test our TableViewer endpoint and our APIGateway endpoint, respectively.
```

{{</highlight>}}

Now we can add those values to our actions in `lib/pipeline-stack.ts` by getting the `stackOutput` of our pipeline stack:
{{<highlight ts "hl_lines=6 16">}}
    // CODE HERE...
    deployStage.addPost(
            new CodeBuildStep('TestViewerEndpoint', {
                projectName: 'TestViewerEndpoint',
                envFromCfnOutputs: {
                    ENDPOINT_URL: deploy.hcViewerUrl
                },
                commands: [
                    'curl -Ssf $ENDPOINT_URL'
                ]
                }),
            new CodeBuildStep('TestAPIGatewayEndpoint', {
                projectName: 'TestAPIGatewayEndpoint',
                envFromCfnOutputs: {
                    ENDPOINT_URL: deploy.hcEndpoint
                },
                commands: [
                    'curl -Ssf $ENDPOINT_URL',

>                     'curl -Ssf $ENDPOINT_URL',
>                     'curl -Ssf $ENDPOINT_URL'

Reviewer: ```suggestion
                    'curl -Ssf $ENDPOINT_URL/hello',
                    'curl -Ssf $ENDPOINT_URL/test'
```
